package-res/resources/web/pentaho/lang/events/WillChange.js

```diff
@@ -0,0 +1,63 @@
+/*!
+ * Copyright 2010 - 2016 Pentaho Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+define([
+  "../Event",
+  "../mixins/mixinChangeset"
+], function(Event, mixinChangeset) {
+  "use strict";
+
+  /**
+   * @name WillChange
+   * @memberOf pentaho.lang.events
+   * @description This event is triggered when a property value is changed in the model.
+   * The listeners of `will:change` are allowed to:
+   * - cancel the event
+   * - make changes to the changeset
+   *
+   * @extends pentaho.lang.Event
```

Comment:
Indeed this is a systemic problem. Reshaped the documentation for all event classes.